### PR TITLE
Update handling of invalid values for non-compulsary fields

### PR DESCRIPTION
**Pull request by lipwei1808**
1. Fix hidden exception throwing for invalid dateTime 
2. Change the handling of invalid values in by Parser

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -190,23 +180,60 @@ public static Type parseType(String type) throws ParseException {
...
      * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code category} is invalid.
      */
-    public static Category parseCategory(String category) {
+    public static Category parseCategory(String category) throws ParseException {
         requireNonNull(category);
         String trimmedCategory = category.trim();
+        if (trimmedCategory.isBlank()) {
+            return new Category(trimmedCategory);
```
Comment (elhy1999) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:212]: Does this just make the category blank? Do we want to instead, give a default category like "Others" or "Uncategorized" instead? 😃

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -190,23 +180,60 @@ public static Type parseType(String type) throws ParseException {
...
      * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code location} is invalid.
      */
-    public static Location parseLocation(String location) {
+    public static Location parseLocation(String location) throws ParseException {
         requireNonNull(location);
         String trimmedLocation = location.trim();
+        if (trimmedLocation.isBlank()) {
+            return new Location(trimmedLocation);
```
Comment (elhy1999) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:231]: See the above comment about the empty "Category" clause.

**Review by elhy1999 — DISMISSED** [review]
LGTM. Minor suggestions about the default values of the "Category" and "Location" clauses

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (lipwei1808) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Not really, this part of the code is slightly unintuitive, i have refactored it in my latest change. 
Summary: The handling of empty values is dealt with inside the DateTime class itself. So the Parser class is solely responsible for parsing the arguments. How this argument is dealt it by the DateTime class

File: src/main/java/seedu/address/model/transaction/DateTime.java [hunk 3]
```diff
@@ -29,15 +29,43 @@ public class DateTime {
...
+        if (dateTime.isBlank()) {
+            this.dateTime = LocalDateTime.now(clock);
             return;
         }
         checkArgument(isValidDateTime(dateTime), MESSAGE_CONSTRAINTS);
         this.dateTime = LocalDateTime.parse(dateTime, DATETIME_FORMATTER);
     }
 
+    public LocalDateTime getDateTime() {
+        return this.dateTime;
```
Comment (woojiahao) [inline src/main/java/seedu/address/model/transaction/DateTime.java:66]: Don't need `this`

**Review by woojiahao — APPROVED** [review]
LGTM, this addresses the issue of the DateTime not raising an error yeah?

**Comment by woojiahao** [comment]
Other than `this` nit, everything else LGTM

**Comment by lipwei1808** [comment]
> LGTM, this addresses the issue of the DateTime not raising an error yeah?

Yes this fixes the issue where an invalid date time does not throw an error appropriately to the user.

**Pull request merged**